src/gen.rs

```diff
@@ -455,7 +462,10 @@ fn gen_method_item(
         SelfType::Ref | SelfType::Mut => {
             // The proxy type could be anything in the `Ref` case, and `&mut`
             // or Box in the `Mut` case.
-            quote! { (**self).#name(#args) }
+             match type_param_count {
+              0 => quote! { (*self).#name(#args) },
+              _ => quote! { (*self).#name::#generic_types(#args) }
+            }
```

Comment:
I was thinking about if we should maybe use the explicit form here: `#proxy_ty_param::#name::#generic_types(*self, #args)`. @KodrAus any opinions about this? I think it is more explicit and it should work.

Reply:
Hm, being consistent about always calling methods as `path(args)` whether they're static or instance seems nice to be :+1:

Reply:
@saresend Could you also change this? :)